spec/index.bs

```diff
@@ -7846,130 +7846,210 @@ Issue: Write normative text about timestamp value resets.
-                            1. Make |swapChain| [=invalid=] and return |swapChain|.
+                            1. Return.
                 </div>
-            1. Return |swapChain|.
+            1. Set |this|.{{GPUPresentationContext/[[validConfiguration]]}} to `true`.
         </div>
 
-    : <dfn>getSwapChainPreferredFormat(adapter)</dfn>
+    : <dfn>getPreferredFormat(adapter)</dfn>
     ::
-        Returns an optimal {{GPUTextureFormat}} to use for swap chains with this context and the
-        given device.
+        Returns an optimal {{GPUTextureFormat}} to use with this presentation context and devices
+        created from the given adapter.
 
-        <div algorithm="GPUCanvasContext.getSwapChainPreferredFormat">
-            **Called on:** {{GPUCanvasContext}} this.
+        <div algorithm="GPUPresentationContext.getPreferredFormat">
+            **Called on:** {{GPUPresentationContext}} this.
 
             **Arguments:**
-            <pre class=argumentdef for="GPUCanvasContext/getSwapChainPreferredFormat(adapter)">
-                |adapter|: Adapter the swap chain format should be queried for.
+            <pre class=argumentdef for="GPUPresentationContext/getPreferredFormat(adapter)">
+                |adapter|: Adapter the format should be queried for.
             </pre>
 
             **Returns:** {{GPUTextureFormat}}
 
             <div class=content-timeline>
-                1. Return an optimal {{GPUTextureFormat}} to use when creating a {{GPUSwapChain}}
-                    with the given |adapter|. Must be one of the [=supported swap chain formats=].
+                1. Return an optimal {{GPUTextureFormat}} to use when calling {{GPUPresentationContext/configure()}}
+                    with the given |adapter|. Must be one of the [=supported presentation context formats=].
             </div>
         </div>
+
+    : <dfn>getCurrentTexture()</dfn>
+    ::
+        Get the {{GPUTexture}} that will be composited to the document by the {{GPUPresentationContext}}
+        next.
+
+        <div algorithm="GPUPresentationContext.getCurrentTexture">
+            **Called on:** {{GPUPresentationContext}} |this|.
+
+            **Returns:** {{GPUTexture}}
+
+            1. If |this|.{{GPUPresentationContext/[[currentTexture]]}} is `null`:
+                1. Let |this|.{{GPUPresentationContext/[[currentTexture]]}} be the result of [$allocating a
+                    new presentation context texture$] for |this|.
+            1. Return |this|.{{GPUPresentationContext/[[currentTexture]]}}.
+        </div>
+
+        Note: Developers can expect that the same {{GPUTexture}} object will be returned by every
+        call to {{GPUPresentationContext/getCurrentTexture()}} made within the same frame (i.e. between
+        invocations of [=Update the rendering=]) unless {{GPUPresentationContext/configure()}} is called.
 </dl>
 
-## GPUSwapChainDescriptor ## {#swapchain-descriptor}
+<div algorithm>
```

Comment:
meta: only skimmed these two algorithms as I assume nothing changed meaningfully

Reply:
You're more-or-less correct in that assumption. The only meaningful change is the check for a non-null `[[descriptor]]` and subsequent error throw at [1]. (Which, now that I look at it, that specific line is probably better lifted into the `getCurrentTexture()` algorithm.)